org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/util/Util.java

```diff
@@ -3358,4 +3358,13 @@ public static IMethod findMethod(IType type, char[] selector, String[] paramType
 		}
 		return method;
 	}
+
+	public static String getEntryName(String zipfileName, ZipEntry entry) {
+		String entryName = entry.getName();
+		if (!java.nio.file.Path.of(zipfileName, entryName).normalize()
```

Comment:
I was confused at first how `Path` would be able to handle zip entries, but you're just using the `Path` API to handle normalization and `startsWith()` in a semantically correct way (not `String.startsWith()`).

Neat trick! Looks good